nixos/modules/services/system/cachix-agent/default.nix

```diff
@@ -0,0 +1,56 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cachix-agent;
+in {
+  meta.maintainers = [ lib.maintainers.domenkozar ];
+
+  options.services.cachix-agent = {
+    enable = mkEnableOption "Cachix Deploy Agent: https://docs.cachix.org/deploy/";
+
+    name = mkOption {
+      type = types.str;
+      description = "Agent name, usually same as the hostname";
+      default = config.networking.hostName;
+    };
+
+    profile = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Profile name, defaults to 'system' (NixOS).";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.cachix;
+      defaultText = literalExpression "pkgs.cachix";
+      description = "Cachix Client package to use.";
+    };
+
+    credentialsFile = mkOption {
+      type = types.path;
+      default = "/etc/cachix-agent.token";
+      description = ''
+        Required file that needs to contain CACHIX_AGENT_TOKEN=...
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.cachix-agent = {
+      description = "Cachix Deploy Agent";
+      after = ["network.target"];
+      path = [ config.nix.package ];
+      wantedBy = [ "multi-user.target" ];
+      # don't restart while changing
+      reloadIfChanged = true;
+      serviceConfig = {
+        Restart = "on-failure";
+        EnvironmentFile = cfg.credentialsFile;
+        ExecStart = "${cfg.package}/bin/cachix deploy agent ${cfg.name} ${if cfg.profile != null then profile else ""}";
```

Comment:
name and profile might want to be escaped (see #154113 / #154123). might even make sense to turn this into a `script` and use the systemd credentials mechanism to pull in the agent token.

Reply:
Both of these options only allow for alphanumberic chars, so I'm not sure this applies :)

About systemd credentials, I'm happy to add support later on.

Reply:
would say it still applies because the options are typed as plain `str`. if they're limited like that we'd also skip escaping (but change the types of the options to `strMatching "[0-9a-zA-Z]+"` or something similar, just to be nice to users and catch errors early. personal opinion 🙂).

Reply:
It's very unlikely someone will hit a problem by using the agent in good faith :)

Reply:
true! was thinking about accidental misconfiguration, not bad faith. but as you say it's unlikely and will just fail in a different manner, so it's not really important here. :)